core/pfe-core/controllers/listbox-controller.ts

```diff
@@ -0,0 +1,518 @@
+import type { ReactiveController, ReactiveControllerHost } from 'lit';
+import { InternalsController } from '@patternfly/pfe-core/controllers/internals-controller.js';
+import { RovingTabindexController } from '@patternfly/pfe-core/controllers/roving-tabindex-controller.js';
+
+/**
+ * Whether list items are arranged vertically or horizontally;
+ * limits arrow keys based on orientation
+ */
+export type ListboxOrientation = 'undefined' | 'horizontal' | 'vertical';
+
+/**
+ * Filtering, multiselect, and orientation options for listbox
+ */
+export interface ListboxConfigOptions {
+  caseSensitive?: boolean;
+  matchAnywhere?: boolean;
+  multi?: boolean;
+  orientation?: ListboxOrientation;
+  getHTMLElement?(): HTMLElement;
+}
+
+export interface ListboxOptionElement extends HTMLElement {
+  value: unknown;
+  selected?: boolean;
+  posInSet?: number;
+  setSize?: number;
+  active?: boolean;
+}
```

Comment:
```suggestion
export interface ListboxOptionElement extends HTMLElement {
  value: unknown;
  selected?: boolean;
  active?: boolean;
}
```

Ideally we could get rid of this whole thing, and maybe take some defaulted change callbacks in the controller init, so that the host is in charge of setting stuff on children.

but if we can't do that, let's at least get rid of these
```ts
  posInSet?: number;
  setSize?: number;
```

and make those controller options instead

Reply:
@bennypowers we'd have to make a separate listbox option controller for that as those props are on listbox options not on the listbox itself.

Reply:
resolved by 79b33a2c